samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java

```diff
@@ -0,0 +1,375 @@
+
+    EventHubClientManager eventHubClientManager = streamEventHubManagers.get(ssp.getStream());
+    String offset = streamPartitionOffsets.get(ssp);
+    Integer partitionId = ssp.getPartition().getPartitionId();
+    String consumerGroup = config.getStreamConsumerGroup(ssp.getSystem(), ssp.getStream());
+
+    try {
+      PartitionReceiver receiver = eventHubClientManager.getEventHubClient()
+              .createReceiverSync(consumerGroup, partitionId.toString(), offset,
+                      !offset.equals(EventHubSystemConsumer.START_OF_STREAM));
+
+      // Timeout for EventHubClient receive
+      receiver.setReceiveTimeout(DEFAULT_EVENTHUB_RECEIVER_TIMEOUT);
+      receiver.setReceiveHandler(streamPartitionHandlers.get(ssp));
+      streamPartitionReceivers.put(ssp, receiver);
+
+    } catch (ServiceBusException e) {
+      throw new SamzaException(
+              String.format("Failed to recreate receiver after ReceiverHandlerError for EventHubs for SystemStreamPartition=%s",
+                      ssp), e);
+    }
+  }
+
+  @Override
+  public void stop() {
+    LOG.debug("Stopping event hub system consumer...");
+    List<CompletableFuture<Void>> futures = new ArrayList<>();
+    streamPartitionReceivers.values().forEach((receiver) -> futures.add(receiver.close()));
+    CompletableFuture<Void> future = CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
+    try {
+      future.get(DEFAULT_SHUTDOWN_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+    } catch (ExecutionException | InterruptedException | TimeoutException e) {
+      LOG.warn("Failed to close receivers", e);
+    }
+    streamEventHubManagers.values().forEach(ehClientManager -> ehClientManager.close(DEFAULT_SHUTDOWN_TIMEOUT_MILLIS));
+  }
+
+  @Override
+  protected BlockingQueue<IncomingMessageEnvelope> newBlockingQueue() {
+    return new LinkedBlockingQueue<>(config.getConsumerBufferCapacity(systemName));
+  }
+
+  protected class PartitionReceiverHandlerImpl extends PartitionReceiveHandler {
+
+    private final Counter eventReadRate;
+    private final Counter eventByteReadRate;
+    private final SamzaHistogram readLatency;
+    private final Counter errorRate;
+    private final Serde<byte[]> serde;
+    SystemStreamPartition ssp;
+
+    PartitionReceiverHandlerImpl(SystemStreamPartition ssp, Counter eventReadRate, Counter eventByteReadRate,
+                                 SamzaHistogram readLatency, Counter readErrors, Serde<byte[]> serde) {
+      super(MAX_EVENT_COUNT_PER_PARTITION_POLL);
+      this.ssp = ssp;
+      this.eventReadRate = eventReadRate;
+      this.eventByteReadRate = eventByteReadRate;
+      this.readLatency = readLatency;
+      errorRate = readErrors;
+      this.serde = serde;
+    }
+
+    @Override
+    public void onReceive(Iterable<EventData> events) {
+      if (events != null) {
+        events.forEach(event -> {
+            byte[] decryptedBody = event.getBytes();
```

Comment:
Remove all references to encryption/decryption. Can use a message interceptor instead